app/helpers/cable_ready_helper.rb

```diff
@@ -4,8 +4,20 @@ module CableReadyHelper
   include CableReady::Compoundable
   include CableReady::StreamIdentifier
 
-  def stream_from(*keys)
+  def stream_from(*keys, html_options: {})
+    tag.stream_from(**build_options(*keys, html_options))
+  end
+
+  def broadcast_from(*keys, url: nil, html_options: {}, &block)
+    options = build_options(*keys, html_options)
+    options[:url] = url if url
+    tag.broadcast_from(**options) { capture(&block) }
+  end
```

Comment:
These are wildly different mechanisms with separate goals and unrelated functionality.

The only thing they have in common is that they are both web components.

Reply:
I agree with @leastbad but I think this boils down to proper naming.

I.e., should it be named `broadcast_from` or something more... "pointed"?

Reply:
Perhaps a more descriptive name would address my concern along with very clear guidance in the docs about each use case and why/when to choose each strategy.

Reply:
especially because it's actually not broadcasting (but that's nitpicking)

I might suggest something a bit up the abstraction ladder, i.e. not describing technicalities but behavior... `morph_updates_for` ...

Reply:
:shipit: | stream_from | broadcast_from
-- | -- | --
Big idea | Explicit | Implicit
Invocation | Method | Callback
Framework | ActionCable | ActionDispatch
Mechanism | WebSockets | Ajax fetch
Operations | All | morphdom
Display | none | block
Content | N/A | HTML fragment
Called from | Anywhere | ActiveModel or PORO

Reply:
> I might suggest something a bit up the abstraction ladder, i.e. not describing technicalities but behavior

`update-from-model`
`watch-model`
or, maybe even
`model-callback`

Reply:
`<fetch-from>` ?

Reply:
@andrewculver

Reply:
Coming back to it, I actually think I like Julian's suggestion of `morph_updates_for` best at this point, and propose we move forward with that unless someone has something so shockingly great that we can't not use it.

But let's wrap it up, too. 😉

Reply:
I like `enable_ui_updates` and `broadcast_ui_updates` for the model.

Then, instead of `auto_update`, why not `updates_for` or `updates_with`? **This succeeds at not only "what not how", but is also conceptually similar to Rails' `form_for` or `form_with`.** You get that nice, English-esque cadence.

Reply:
`auto_updates_for`👌🏻

Reply:
In the view, the best I can think to suggest is the following. It also accurately describes what is actually happening.

```
<%= updates_from current_user do %>
  ...
<% end %>
```

Works well for a collection, as well:

```
<%= updates_from @team, :some_collection do %>
  ...
<% end %>
```

Reply:
I think `_for` aligns better with the `_for` nomenclature for resources:

- form_for
- stream_for
- fields_for

Reply:
I take it back `updates_from` feels like a misnomer, because it doesn't actually update from the specified model or collection, it updates _because of_ those things... maybe you were right with `updates_for`, @leastbad. It's accurate.

Reply:
`auto_updates_for` also a 👌 suggestion.

Reply:
`enable_ui_updates` and `broadcast_ui_updates` doesn't feel quite right to me. We're not _actually_ broadcasting the UI updates. I don't have a better suggestion though, without just reverting back to `enable_broadcasting` (or something similarly simple) and `broadcast: true`...

Reply:
Less concerned with `enable_ui_updates`, but would prefer for the invocation on collections to be short and sweet if possible.

Reply:
I hear you, but I also want to ensure that the developer has the correct mental model driven by the semantics we choose.

Reply:
Agreed. I would keep it at `broadcast: true`. Maybe `broadcast_updates`

Reply:
I'd also like to avoid potential naming conflicts that Hotwire may choose to co-opt in the future. Hard to predict, but generic names like `broadcast` have a high probability of being taken from us at some point.

Reply:
Not to complicate things even further but I think it would be good to also have some symmetry between the model code and the view code.

For me I think something like `stream_updates_for @post` for the view layer and `stream_updates: true` in the model layer would be nice. If not those words exactly just something symmetrical at least so it indicates where it's coming from and going to. In my head it's like a pipe (the Mario kind) and it the data goes in at the model layer and comes out at the view layer.

Reply:
This is a great point. It would be ideal to grep the project and find both the model and view template matches.